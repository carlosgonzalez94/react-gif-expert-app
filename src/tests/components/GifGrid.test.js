import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test of the component <GifGridItem />', () => {
    let category = 'One Punch';
    let wrapper;

    test('Should render the component AddCategory correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Should render items when the images are loaded', () => {
        //coloco la respuesta que se simulara
        const gifs = [
            {
                id: 'fake id',
                title: 'fake title',
                url: 'fake url'
            }
        ];

        //mockeo la respuesta de la funci√≥n
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        //renderizo el componente
        wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
    
})
