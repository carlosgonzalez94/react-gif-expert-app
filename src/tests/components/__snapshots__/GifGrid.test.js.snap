// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the component <GifGridItem /> Should render items when the images are loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="fake id"
      key="fake id"
      title="fake title"
      url="fake url"
    />
  </div>
</Fragment>
`;

exports[`Test of the component <GifGridItem /> Should render the component AddCategory correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
